### PR TITLE
update mocha and istanbul package dependencies

### PR DESCRIPTION
**Pull request by balajikris**
The content to be added to Changelog is as follows:
- update mocha and instanbul package dependencies.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@balajikris__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (balajik). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by balajikris** [comment]
I'll wait to see if the CI passes and for someone from the team to comment if this would be an issue in either debugging tests or running in the CI queue. I notice we were staying in a old version of `mocha`, but I don't know why. I'm hoping the CI run on this PR would tell me.

This PR addresses the following npm warnings:

~~~
npm WARN deprecated minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
~~~

**Comment by balajikris** [comment]
Also, should I add myself to a contributors list somewhere so that the Azure PR bot does not drop by on every PR I make?

File: package.json [hunk 1]
```diff
@@ -114,8 +114,8 @@
     "read": "^1.0.7"
   },
   "devDependencies": {
-    "istanbul": "0.3.19",
-    "mocha": "1.16.0",
+    "istanbul": "0.4.5",
+    "mocha": "3.0.2",
```
Comment (balajikris) [inline package.json]: if it works with this, should I relax this a little and accept any minor version changes with ^ at the beginning for both packages?

**Comment by balajikris** [comment]
okay, looks like the upgrade failed tests, as i kinda expected. Will investigate and see if this is feasible.

**Comment by balajikris** [comment]
made progress here. it turns out that mochaSuiteObject's currentTest property is null in `before` hook of a suite. We then error out trying to get fullTitle() from currentTest object. This happens when upgrading to latest version of mocha but not in earlier versions. Needs further investigation as to why `currentTest` would be null.

**Pull request closed without merging**